# Merging baseline, visit 24, visit 26 datasets together

In [1]:
import pandas as pd
import numpy as np


In [29]:
impact_baseline = pd.read_excel("Data/IMPACT_Study/cleaned_datasets/impact_all_baseline_clean_final.xlsx")
impact_24 = pd.read_excel("Data/IMPACT_Study/cleaned_datasets/impact_all_24_clean_final.xlsx")
impact_26 = pd.read_excel("Data/IMPACT_Study/cleaned_datasets/impact_all_26_clean_final.xlsx")



In [30]:
print(impact_baseline.shape)
print(len(set(impact_baseline['Participant ID'].unique())))
impact_baseline.head()

(141, 22)
141


Participant ID Collection Date  Visit   Age  Male  White  Black  Asian  \
0  IMPACT_101655      2012-10-11     -2  45.6     1      1      0      0   
1  IMPACT_102436      2013-03-23     -2  46.8     1      1      0      0   
2  IMPACT_105670      2013-04-04     -2  36.0     0      1      0      0   
3  IMPACT_113135      2013-11-25     -2  31.2     0      1      0      0   
4  IMPACT_115876      2014-03-19     -2  26.4     1      0      0      0   

   Other  Mixed  ...  Flare (mm)  Total IgE (kU/L)  Peanut IgE (kU/L)  \
0    NaN      0  ...         NaN              92.0               25.7   
1    NaN      0  ...         NaN             187.0               36.2   
2    NaN      0  ...         NaN             118.0               51.3   
3    NaN      0  ...         NaN             408.0              241.0   
4    NaN      1  ...         NaN             344.0               19.8   

   Ara h1 (kU/L)  Ara h2 (kU/L)  Ara h3 (kU/L)  Ara h6 (kUA/L)  Ara h8 (kU/L)  \
0           8.76           27.5           0.50             NaN            NaN   
1           2.34           48.1           0.24             NaN            NaN   
2           2.73           61.6           2.40             NaN            NaN   
3           6.45          116.0           0.79             NaN            NaN   
4          12.10           13.2           0.24             NaN            NaN   

   Ara h9 (kU/L)  OFC Pass  
0            NaN         1  
1            NaN         1  
2            NaN         1  
3            NaN         1  
4            NaN         1  

[5 rows x 22 columns]

In [31]:
print(impact_24.shape)
print(len(set(impact_24['Participant ID'].unique())))
impact_24.head()

(115, 22)
115


Participant ID Collection Date  Visit   Age  Male  White  Black  Asian  \
0  IMPACT_102436      2015-11-20     24  79.5     1      1      0      0   
1  IMPACT_105670      2015-11-25     24  68.2     0      1      0      0   
2  IMPACT_113135      2016-09-12     24  65.3     0      1      0      0   
3  IMPACT_115876      2016-11-09     24  58.6     1      0      0      0   
4  IMPACT_136775      2017-11-20     24  74.3     0      0      0      0   

   Other  Mixed  ...  Flare (mm)  Total IgE (kU/L)  Peanut IgE (kU/L)  \
0    NaN      0  ...         NaN                97               8.41   
1    NaN      0  ...         NaN                70              10.70   
2    NaN      0  ...         NaN               565             243.00   
3    NaN      1  ...         NaN               727              19.20   
4    NaN      1  ...         NaN                25               9.06   

   Ara h1 (kU/L)  Ara h2 (kU/L)  Ara h3 (kU/L)  Ara h6 (kUA/L)  Ara h8 (kU/L)  \
0           1.14           8.67           0.05             NaN            NaN   
1           1.95          10.40           0.39             NaN            NaN   
2          16.30         181.00           1.40             NaN            NaN   
3           8.75           9.54           0.05             NaN            NaN   
4           0.88           7.80           0.05             NaN            NaN   

   Ara h9 (kU/L)  OFC Pass  
0            NaN         1  
1            NaN         1  
2            NaN         0  
3            NaN         0  
4            NaN         0  

[5 rows x 22 columns]

In [32]:
print(impact_26.shape)
print(len(set(impact_26['Participant ID'].unique())))
impact_26.head()

(93, 22)
93


Participant ID Collection Date  Visit   Age  Male  White  Black  Asian  \
0  IMPACT_102436      2016-05-17     26  85.5     1      1      0      0   
1  IMPACT_105670      2016-05-27     26  74.3     0      1      0      0   
2  IMPACT_115876      2017-05-10     26  64.7     1      0      0      0   
3  IMPACT_139237      2018-02-22     26  75.7     1      0      0      1   
4  IMPACT_196748      2017-08-13     26  82.9     1      1      0      0   

   Other  Mixed  ...  Flare (mm)  Total IgE (kU/L)  Peanut IgE (kU/L)  \
0    NaN      0  ...         NaN              80.0              15.30   
1    NaN      0  ...         NaN              68.0               8.92   
2    NaN      1  ...         NaN            1215.0              16.40   
3    NaN      0  ...         NaN             490.0             148.00   
4    NaN      0  ...         NaN              52.0               0.45   

   Ara h1 (kU/L)  Ara h2 (kU/L)  Ara h3 (kU/L)  Ara h6 (kUA/L)  Ara h8 (kU/L)  \
0            NaN            NaN            NaN             NaN            NaN   
1           3.94           8.24           0.32             NaN            NaN   
2           7.45           8.99           0.05             NaN            NaN   
3          20.30         112.00           2.16             NaN            NaN   
4           0.05           0.37           0.05             NaN            NaN   

   Ara h9 (kU/L)  OFC Pass  
0            NaN         0  
1            NaN         0  
2            NaN         0  
3            NaN         0  
4            NaN         0  

[5 rows x 22 columns]

In [33]:
# concatinating the DFs
impact_baseline_24 = pd.concat([impact_baseline, impact_24], ignore_index=True)

In [34]:
print(impact_baseline_24.shape)
impact_baseline_24.head()

(256, 22)


Participant ID Collection Date  Visit   Age  Male  White  Black  Asian  \
0  IMPACT_101655      2012-10-11     -2  45.6     1      1      0      0   
1  IMPACT_102436      2013-03-23     -2  46.8     1      1      0      0   
2  IMPACT_105670      2013-04-04     -2  36.0     0      1      0      0   
3  IMPACT_113135      2013-11-25     -2  31.2     0      1      0      0   
4  IMPACT_115876      2014-03-19     -2  26.4     1      0      0      0   

   Other  Mixed  ...  Flare (mm)  Total IgE (kU/L)  Peanut IgE (kU/L)  \
0    NaN      0  ...         NaN              92.0               25.7   
1    NaN      0  ...         NaN             187.0               36.2   
2    NaN      0  ...         NaN             118.0               51.3   
3    NaN      0  ...         NaN             408.0              241.0   
4    NaN      1  ...         NaN             344.0               19.8   

   Ara h1 (kU/L)  Ara h2 (kU/L)  Ara h3 (kU/L)  Ara h6 (kUA/L)  Ara h8 (kU/L)  \
0           8.76           27.5           0.50             NaN            NaN   
1           2.34           48.1           0.24             NaN            NaN   
2           2.73           61.6           2.40             NaN            NaN   
3           6.45          116.0           0.79             NaN            NaN   
4          12.10           13.2           0.24             NaN            NaN   

   Ara h9 (kU/L)  OFC Pass  
0            NaN         1  
1            NaN         1  
2            NaN         1  
3            NaN         1  
4            NaN         1  

[5 rows x 22 columns]

In [35]:
impact_baseline_24_26 = pd.concat([impact_baseline_24, impact_26], ignore_index=True)


In [36]:
print(impact_baseline_24_26.shape)
impact_baseline_24_26.head()

(349, 22)


Participant ID Collection Date  Visit   Age  Male  White  Black  Asian  \
0  IMPACT_101655      2012-10-11     -2  45.6     1      1      0      0   
1  IMPACT_102436      2013-03-23     -2  46.8     1      1      0      0   
2  IMPACT_105670      2013-04-04     -2  36.0     0      1      0      0   
3  IMPACT_113135      2013-11-25     -2  31.2     0      1      0      0   
4  IMPACT_115876      2014-03-19     -2  26.4     1      0      0      0   

   Other  Mixed  ...  Flare (mm)  Total IgE (kU/L)  Peanut IgE (kU/L)  \
0    NaN      0  ...         NaN              92.0               25.7   
1    NaN      0  ...         NaN             187.0               36.2   
2    NaN      0  ...         NaN             118.0               51.3   
3    NaN      0  ...         NaN             408.0              241.0   
4    NaN      1  ...         NaN             344.0               19.8   

   Ara h1 (kU/L)  Ara h2 (kU/L)  Ara h3 (kU/L)  Ara h6 (kUA/L)  Ara h8 (kU/L)  \
0           8.76           27.5           0.50             NaN            NaN   
1           2.34           48.1           0.24             NaN            NaN   
2           2.73           61.6           2.40             NaN            NaN   
3           6.45          116.0           0.79             NaN            NaN   
4          12.10           13.2           0.24             NaN            NaN   

   Ara h9 (kU/L)  OFC Pass  
0            NaN         1  
1            NaN         1  
2            NaN         1  
3            NaN         1  
4            NaN         1  

[5 rows x 22 columns]

# Checking the number of participants 
- should be 141 but there are 143!

In [37]:
len(set(impact_baseline_24_26['Participant ID'].unique()))

143

# Investigating why there are more unique participant IDs than we started with in the baseline dataset

In [38]:
list_baseline = impact_baseline['Participant ID'].tolist()
list_24 = impact_24['Participant ID'].tolist()
list_26 = impact_26['Participant ID'].tolist()

common_participants = set(list_baseline).intersection(list_24, list_26)
common_participants = list(common_participants)


In [39]:
len(set(common_participants))

91

In [40]:
common_participants = list_baseline+list_24+list_26
common_participants = list(set(common_participants))
len(set(common_participants))

143

# Manually looking at original datasets shows that there are some participants who did not follow the chronological order of visits
Example: 
- some are missing visit 24 but have visit -2, 16, and 26
- 2 are missing baseline visits and did not get added to the baseline dataset

In [41]:
# sorting dataset by participant ID and Visit

# Assuming you have a dataframe 'impact_baseline_24_26' and you want to sort it based on 'Participant ID' and 'Visit'
impact_combined_all = impact_baseline_24_26.sort_values(by=['Participant ID', 'Visit'], ascending=[True, True])

# Reset the index after sorting
impact_combined_all = impact_combined_all.reset_index(drop=True)



In [42]:
print(impact_combined_all.shape)
impact_combined_all.head(10)

(349, 22)


Participant ID Collection Date  Visit   Age  Male  White  Black  Asian  \
0  IMPACT_101655      2012-10-11     -2  45.6     1      1      0      0   
1  IMPACT_102436      2013-03-23     -2  46.8     1      1      0      0   
2  IMPACT_102436      2015-11-20     24  79.5     1      1      0      0   
3  IMPACT_102436      2016-05-17     26  85.5     1      1      0      0   
4  IMPACT_105670      2013-04-04     -2  36.0     0      1      0      0   
5  IMPACT_105670      2015-11-25     24  68.2     0      1      0      0   
6  IMPACT_105670      2016-05-27     26  74.3     0      1      0      0   
7  IMPACT_113135      2013-11-25     -2  31.2     0      1      0      0   
8  IMPACT_113135      2016-09-12     24  65.3     0      1      0      0   
9  IMPACT_115876      2014-03-19     -2  26.4     1      0      0      0   

   Other  Mixed  ...  Flare (mm)  Total IgE (kU/L)  Peanut IgE (kU/L)  \
0    NaN      0  ...         NaN              92.0              25.70   
1    NaN      0  ...         NaN             187.0              36.20   
2    NaN      0  ...         NaN              97.0               8.41   
3    NaN      0  ...         NaN              80.0              15.30   
4    NaN      0  ...         NaN             118.0              51.30   
5    NaN      0  ...         NaN              70.0              10.70   
6    NaN      0  ...         NaN              68.0               8.92   
7    NaN      0  ...         NaN             408.0             241.00   
8    NaN      0  ...         NaN             565.0             243.00   
9    NaN      1  ...         NaN             344.0              19.80   

   Ara h1 (kU/L)  Ara h2 (kU/L)  Ara h3 (kU/L)  Ara h6 (kUA/L)  Ara h8 (kU/L)  \
0           8.76          27.50           0.50             NaN            NaN   
1           2.34          48.10           0.24             NaN            NaN   
2           1.14           8.67           0.05             NaN            NaN   
3            NaN            NaN            NaN             NaN            NaN   
4           2.73          61.60           2.40             NaN            NaN   
5           1.95          10.40           0.39             NaN            NaN   
6           3.94           8.24           0.32             NaN            NaN   
7           6.45         116.00           0.79             NaN            NaN   
8          16.30         181.00           1.40             NaN            NaN   
9          12.10          13.20           0.24             NaN            NaN   

   Ara h9 (kU/L)  OFC Pass  
0            NaN         1  
1            NaN         1  
2            NaN         1  
3            NaN         0  
4            NaN         1  
5            NaN         1  
6            NaN         0  
7            NaN         1  
8            NaN         0  
9            NaN         1  

[10 rows x 22 columns]

In [43]:
# commenting out to prevent export again 

# exporting combined dataset 
# impact_combined_all.to_excel('Data/IMPACT_Study/impact_combined_all_clean_final.xlsx', index=False)
